File: 007.reverse-integer/007.reverse-integer.java
class Solution {
    public int reverse(int c) {
            
  int temp = 0;
  int rev=0;
  while (c != 0) {
   rev = rev * 10 + c%10;
   if(temp!=rev/10){
    return 0;
   }
   temp =rev;
   c = c/10;
  }
  return rev;
 
    }
}

File: 007.reverse-integer/007.reverse-integer.py
class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x == -2 ** 31:
            return 0
        reversestr = ''
        if x < 0:
            reversestr = '-'
        x = abs(x)
        string = str(x)
        for i in range(len(string) - 1, -1, -1):
            if i == len(string) - 1 and string[i] == 0:
                continue
            reversestr += string[i]
        x = int(reversestr)
        if x < -2 ** 31 or x > 2 ** 31 - 1:
            return 0
        else:
            return x
x = 1534236469
sol = Solution()
print(sol.reverse(x))
